packages/esm-patient-registration-app/src/patient-registration/patient-registration.resource.tsx

```diff
@@ -182,11 +182,20 @@ export function usePatientPhoto(patientUuid: string): UsePatientPhotoResult {
 }
 
 export async function fetchPerson(query: string) {
-  const abortController = new AbortController();
-
-  return openmrsFetch(`/ws/rest/v1/person?q=${query}`, {
-    signal: abortController.signal,
-  });
+  const BASE_API = '/ws/rest/v1';
+  const ctrl = new AbortController();
+  const { signal } = ctrl;
+
+  try {
+    const endpoint = (await openmrsFetch(`${BASE_API}/person?q=${query}`, { signal })).data.results.length
+      ? `${BASE_API}/person?q=${query}`
+      : `${BASE_API}/patient?q=${query}`;
```

Comment:
This looks like something that could be optimized. 2 api calls will end up being made to `/person` with the second one being made  on line 194